journalbeat/input/input.go

```diff
@@ -135,7 +135,11 @@ func (i *Input) Run() {
 	}
 	defer client.Close()
 
-	i.publishAll(client)
+	go i.publishAll(client)
```

Comment:
hmmm.... we start a go-routine which starts a go-routine which runs another go-routine. With this change the Run method only returns early, we potentially leak the `publishAll` go-routine, yet shutdown can continue (even if shutdown is not clean).

`publishAll` will be unblocked from the publisher pipeline when `client.Close()` kicks in. By closing the `Client` on `Stop` we should be able to unblock `publishAll` as well. This is one of the reasons I'd like to pass a beats context. Then we can pass the context to `Connect` and don't need to jump through multiple hoops to get shutdown logic working.